### DISCUSSION TITLE
Rich Text editor image path issue?

### DISCUSSION THREAD

**Discussion by 872342** [pocketbase/pocketbase#1975] [Q&A]
When attaching an image via the rich editor field, you can choose an image or attach a URL.
I believe with choosing an image, they embed as a base64 string, and using a URL just sets the img src to that URL.
I think a good feature could be if the source selector in the rich editor allowed you to upload a file into a collection from there, or choose from one of your already uploaded files, maybe from a "files" collection.

When I manually upload an image to a pocketbase collection, then link the image path source in the rich editor, my source gets shortened to "../api/files/" (removing the domain part), which 404s when the website domain is different from the pocketbase domain. 
There doesn't seem to be a way to force the source to remain as the full URL? 

Thanks

**Comment by ganigeorgiev** [comment]
> I think a good feature could be if the source selector in the rich editor allowed you to upload a file into a collection from there, or choose from one of your already uploaded files, maybe from a "files" collection.

This is hard to implement and generalize because collections could have other fields and constraints and you may not be able to create a record just uploading a file.

---

> There doesn't seem to be a way to force the source to remain as the full URL?

There is TinyMCE configuration for that - https://www.tiny.cloud/docs/tinymce/6/url-handling/#relative_urls

I'll consider updating it sometime later today and will push a new minor release.

**Reply by ganigeorgiev** [reply]
Actually, on second thought, changing the default TinyMCE `relative_urls` behavior may not be a good idea since it could cause issues when migrating between environments.

For now as a workaround you can replace the `../api/files/` string in your html with your actual "{domain}/api/files/".

I'll consider making it a configurable field option in the future but for now remains a low priority.

**Comment by hungcrush** [comment]
@ganigeorgiev the best solution is Media Library 😬 
TinyMCE -> Image -> Select file from Media

**Reply by ganigeorgiev** [reply]
Dedicated "Media library/collection" is not planned for now.

Although to some extend it can be used this way, the Admin UI is not really intended to be a  replacement for a CMS.

**Reply by ganigeorgiev** [reply]
_Just for info - "Media library"-like functionality for the `editor` field was added in the `develop` branch. For more info and screenshots of the current UI you can check in https://github.com/pocketbase/pocketbase/discussions/3429#discussioncomment-7172736._
